mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOpsDialect.h

```diff
@@ -34,9 +39,30 @@ class MethodRegistry {
   llvm::Optional<llvm::StringRef> lookupMethod(::mlir::TypeID Type,
                                                llvm::StringRef Name) const;
 
+  /// Add a definition for the given method to be used when lowering
+  /// SYCLMethodOpInterface instances.
+  ///
+  /// A call to this function will fail if \p MayOverride is false and we try to
+  /// override an already present definition.
+  void registerDefinition(llvm::StringRef methodName,
+                          mlir::func::FuncOp Definition,
+                          bool MayOverride = false);
```

Comment:
I propose we remove the ability to override a previously register method, at least until we find a need.
(Unless you have a motivating example for this functionality)

Reply:
This is needed as I said in my previous comment